src/meson.build

```diff
@@ -680,18 +680,31 @@ endif
 if get_option('tests').enabled()
   # TODO: MSVC gives the following,
   # error LNK2019: unresolved external symbol "unsigned __int64 const * const _hb_NullPool"
-  if cpp.get_argument_syntax() != 'msvc'
+  if cpp.get_define('_MSC_FULL_VER') == ''
     noinst_programs = {
       'main': 'main.cc',
+      'test-algs': ['test-algs.cc', 'hb-static.cc'],
```

Comment:
Why do these tests not work with MS compilers?

Reply:
All 13 tests fail with further variations of the LNK2019 error above:

<details><summary>Linker errors</summary>

```
[1/13] Linking target src/test-priority-queue.exe
FAILED: src/test-priority-queue.exe src/test-priority-queue.pdb
"clang"  -Wl,/MACHINE:X64 -Wl,/OUT:src/test-priority-queue.exe src/test-priority-queue.exe.p/test-priority-queue.cc.obj src/test-priority-queue.exe.p/hb-static.cc.obj "-Wl,/nologo" "-Wl,/release" "-Wl,/nologo" "-Wl,/DEBUG" "-Wl,/PDB:src/test-priority-queue.pdb" "src/harfbuzz.lib" "subprojects/freetype-2.13.0/freetype.lib" "-Wl,/SUBSYSTEM:CONSOLE" "-lkernel32" "-luser32" "-lgdi32" "-lwinspool" "-lshell32" 
"-lole32" "-loleaut32" "-luuid" "-lcomdlg32" "-ladvapi32"
test-priority-queue.cc.obj : error LNK2019: unresolved external symbol "unsigned __int64 * _hb_CrapPool" (?_hb_CrapPool@@3PA_KA) referenced in function "struct hb_pair_t<__int64,unsigned int> & __cdecl Crap<struct hb_pair_t<__int64,unsigned int> >(void)" (??$Crap@U?$hb_pair_t@_JI@@@@YAAEAU?$hb_pair_t@_JI@@XZ)
test-priority-queue.cc.obj : error LNK2019: unresolved external symbol "unsigned __int64 const * const _hb_NullPool" (?_hb_NullPool@@3QB_KB) referenced in function "public: static struct hb_pair_t<__int64,unsigned int> const & __cdecl Null<struct hb_pair_t<__int64,unsigned int> >::get_null(void)" (?get_null@?$Null@U?$hb_pair_t@_JI@@@@SAAEBU?$hb_pair_t@_JI@@XZ)
src\test-priority-queue.exe : fatal error LNK1120: 2 unresolved externals
clang: error: linker command failed with exit code 1120 (use -v to see invocation)
[2/13] Linking target src/test-set.exe
FAILED: src/test-set.exe src/test-set.pdb
"clang"  -Wl,/MACHINE:X64 -Wl,/OUT:src/test-set.exe src/test-set.exe.p/test-set.cc.obj src/test-set.exe.p/hb-static.cc.obj "-Wl,/nologo" "-Wl,/release" "-Wl,/nologo" "-Wl,/DEBUG" "-Wl,/PDB:src/test-set.pdb" "src/harfbuzz.lib" "subprojects/freetype-2.13.0/freetype.lib" "-Wl,/SUBSYSTEM:CONSOLE" "-lkernel32" "-luser32" "-lgdi32" "-lwinspool" "-lshell32" "-lole32" "-loleaut32" "-luuid" "-lcomdlg32" "-ladvapi32"
test-set.cc.obj : error LNK2019: unresolved external symbol "unsigned __int64 * _hb_CrapPool" (?_hb_CrapPool@@3PA_KA) referenced in function "struct hb_user_data_array_t::hb_user_data_item_t & __cdecl Crap<struct hb_user_data_array_t::hb_user_data_item_t>(void)" (??$Crap@Uhb_user_data_item_t@hb_user_data_array_t@@@@YAAEAUhb_user_data_item_t@hb_user_data_array_t@@XZ)
test-set.cc.obj : error LNK2019: unresolved external symbol "unsigned __int64 const * const _hb_NullPool" (?_hb_NullPool@@3QB_KB) referenced in function "public: static struct hb_user_data_array_t::hb_user_data_item_t const & __cdecl Null<struct hb_user_data_array_t::hb_user_data_item_t>::get_null(void)" (?get_null@?$Null@Uhb_user_data_item_t@hb_user_data_array_t@@@@SAAEBUhb_user_data_item_t@hb_user_data_array_t@@XZ)
src\test-set.exe : fatal error LNK1120: 2 unresolved externals
clang: error: linker command failed with exit code 1120 (use -v to see invocation)
[3/13] Linking target src/test-classdef-graph.exe
FAILED: src/test-classdef-graph.exe src/test-classdef-graph.pdb
"clang"  -Wl,/MACHINE:X64 -Wl,/OUT:src/test-classdef-graph.exe src/test-classdef-graph.exe.p/graph_test-classdef-graph.cc.obj src/test-classdef-graph.exe.p/hb-static.cc.obj src/test-classdef-graph.exe.p/graph_gsubgpos-context.cc.obj "-Wl,/nologo" "-Wl,/release" "-Wl,/nologo" "-Wl,/DEBUG" "-Wl,/PDB:src/test-classdef-graph.pdb" "src/harfbuzz.lib" "subprojects/freetype-2.13.0/freetype.lib" "-Wl,/SUBSYSTEM:CONSOLE" "-lkernel32" "-luser32" "-lgdi32" "-lwinspool" "-lshell32" "-lole32" "-loleaut32" "-luuid" "-lcomdlg32" "-ladvapi32"
graph_test-classdef-graph.cc.obj : error LNK2019: unresolved external symbol "unsigned __int64 * _hb_CrapPool" (?_hb_CrapPool@@3PA_KA) referenced in function "struct hb_user_data_array_t::hb_user_data_item_t & __cdecl Crap<struct hb_user_data_array_t::hb_user_data_item_t>(void)" (??$Crap@Uhb_user_data_item_t@hb_user_data_array_t@@@@YAAEAUhb_user_data_item_t@hb_user_data_array_t@@XZ)
graph_gsubgpos-context.cc.obj : error LNK2001: unresolved external symbol "unsigned __int64 * _hb_CrapPool" (?_hb_CrapPool@@3PA_KA)
graph_test-classdef-graph.cc.obj : error LNK2019: unresolved external symbol "unsigned __int64 const * const _hb_NullPool" (?_hb_NullPool@@3QB_KB) referenced in function "public: static struct hb_user_data_array_t::hb_user_data_item_t const & __cdecl Null<struct hb_user_data_array_t::hb_user_data_item_t>::get_null(void)" (?get_null@?$Null@Uhb_user_data_item_t@hb_user_data_array_t@@@@SAAEBUhb_user_data_item_t@hb_user_data_array_t@@XZ)
graph_gsubgpos-context.cc.obj : error LNK2001: unresolved external symbol "unsigned __int64 const * const _hb_NullPool" (?_hb_NullPool@@3QB_KB)
src\test-classdef-graph.exe : fatal error LNK1120: 2 unresolved externals
clang: error: linker command failed with exit code 1120 (use -v to see invocation)
[4/13] Linking target src/test-algs.exe
FAILED: src/test-algs.exe src/test-algs.pdb
"clang"  -Wl,/MACHINE:X64 -Wl,/OUT:src/test-algs.exe src/test-algs.exe.p/test-algs.cc.obj src/test-algs.exe.p/hb-static.cc.obj "-Wl,/nologo" "-Wl,/release" "-Wl,/nologo" "-Wl,/DEBUG" "-Wl,/PDB:src/test-algs.pdb" "src/harfbuzz.lib" "subprojects/freetype-2.13.0/freetype.lib" "-Wl,/SUBSYSTEM:CONSOLE" "-lkernel32" "-luser32" "-lgdi32" "-lwinspool" "-lshell32" "-lole32" "-loleaut32" "-luuid" "-lcomdlg32" "-ladvapi32"
test-algs.cc.obj : error LNK2019: unresolved external symbol "unsigned __int64 const * const _hb_NullPool" (?_hb_NullPool@@3QB_KB) referenced in function "public: static unsigned __int64 const & __cdecl Null<unsigned __int64>::get_null(void)" (?get_null@?$Null@_K@@SAAEB_KXZ)
test-algs.cc.obj : error LNK2019: unresolved external symbol "unsigned __int64 * _hb_CrapPool" (?_hb_CrapPool@@3PA_KA) referenced in function "struct hb_user_data_array_t::hb_user_data_item_t & __cdecl Crap<struct hb_user_data_array_t::hb_user_data_item_t>(void)" (??$Crap@Uhb_user_data_item_t@hb_user_data_array_t@@@@YAAEAUhb_user_data_item_t@hb_user_data_array_t@@XZ)
src\test-algs.exe : fatal error LNK1120: 2 unresolved externals
clang: error: linker command failed with exit code 1120 (use -v to see invocation)
[5/13] Linking target src/test-iter.exe
FAILED: src/test-iter.exe src/test-iter.pdb
"clang"  -Wl,/MACHINE:X64 -Wl,/OUT:src/test-iter.exe src/test-iter.exe.p/test-iter.cc.obj src/test-iter.exe.p/hb-static.cc.obj "-Wl,/nologo" "-Wl,/release" "-Wl,/nologo" "-Wl,/DEBUG" "-Wl,/PDB:src/test-iter.pdb" "src/harfbuzz.lib" "subprojects/freetype-2.13.0/freetype.lib" "-Wl,/SUBSYSTEM:CONSOLE" "-lkernel32" "-luser32" "-lgdi32" "-lwinspool" "-lshell32" "-lole32" "-loleaut32" "-luuid" "-lcomdlg32" "-ladvapi32"
test-iter.cc.obj : error LNK2019: unresolved external symbol "unsigned __int64 * _hb_CrapPool" (?_hb_CrapPool@@3PA_KA) referenced in function "int & __cdecl Crap<int>(void)" (??$Crap@H@@YAAEAHXZ)
test-iter.cc.obj : error LNK2019: unresolved external symbol "unsigned __int64 const * const _hb_NullPool" (?_hb_NullPool@@3QB_KB) referenced in function "public: static int const & __cdecl Null<int>::get_null(void)" (?get_null@?$Null@H@@SAAEBHXZ)
test-iter.cc.obj : error LNK2019: unresolved external symbol "unsigned int const minus_1" (?minus_1@@3IB) referenced in function "public: static unsigned int const & __cdecl hb_hashmap_t<unsigned int,unsigned int,1>::item_t::default_value<1,0>(void)" (??$default_value@$00$0A@@item_t@?$hb_hashmap_t@II$00@@SAAEBIXZ)
src\test-iter.exe : fatal error LNK1120: 3 unresolved externals
clang: error: linker command failed with exit code 1120 (use -v to see invocation)
[6/13] Linking target src/test-serialize.exe
FAILED: src/test-serialize.exe src/test-serialize.pdb
"clang"  -Wl,/MACHINE:X64 -Wl,/OUT:src/test-serialize.exe src/test-serialize.exe.p/test-serialize.cc.obj src/test-serialize.exe.p/hb-static.cc.obj "-Wl,/nologo" "-Wl,/release" "-Wl,/nologo" "-Wl,/DEBUG" "-Wl,/PDB:src/test-serialize.pdb" "src/harfbuzz.lib" "subprojects/freetype-2.13.0/freetype.lib" "-Wl,/SUBSYSTEM:CONSOLE" "-lkernel32" "-luser32" "-lgdi32" "-lwinspool" "-lshell32" "-lole32" "-loleaut32" "-luuid" "-lcomdlg32" "-ladvapi32"
test-serialize.cc.obj : error LNK2019: unresolved external symbol "unsigned __int64 * _hb_CrapPool" (?_hb_CrapPool@@3PA_KA) referenced in function "struct hb_user_data_array_t::hb_user_data_item_t & __cdecl Crap<struct hb_user_data_array_t::hb_user_data_item_t>(void)" (??$Crap@Uhb_user_data_item_t@hb_user_data_array_t@@@@YAAEAUhb_user_data_item_t@hb_user_data_array_t@@XZ)
test-serialize.cc.obj : error LNK2019: unresolved external symbol "unsigned __int64 const * const _hb_NullPool" (?_hb_NullPool@@3QB_KB) referenced in function "public: static struct hb_user_data_array_t::hb_user_data_item_t const & __cdecl Null<struct hb_user_data_array_t::hb_user_data_item_t>::get_null(void)" (?get_null@?$Null@Uhb_user_data_item_t@hb_user_data_array_t@@@@SAAEBUhb_user_data_item_t@hb_user_data_array_t@@XZ)
src\test-serialize.exe : fatal error LNK1120: 2 unresolved externals
clang: error: linker command failed with exit code 1120 (use -v to see invocation)
[7/13] Linking target src/test-instancer-solver.exe
FAILED: src/test-instancer-solver.exe src/test-instancer-solver.pdb
"clang"  -Wl,/MACHINE:X64 -Wl,/OUT:src/test-instancer-solver.exe src/test-instancer-solver.exe.p/test-subset-instancer-solver.cc.obj src/test-instancer-solver.exe.p/hb-subset-instancer-solver.cc.obj src/test-instancer-solver.exe.p/hb-static.cc.obj "-Wl,/nologo" "-Wl,/release" "-Wl,/nologo" "-Wl,/DEBUG" "-Wl,/PDB:src/test-instancer-solver.pdb" "src/harfbuzz.lib" "subprojects/freetype-2.13.0/freetype.lib" "-Wl,/SUBSYSTEM:CONSOLE" "-lkernel32" "-luser32" "-lgdi32" "-lwinspool" "-lshell32" "-lole32" "-loleaut32" "-luuid" "-lcomdlg32" "-ladvapi32"
test-subset-instancer-solver.cc.obj : error LNK2019: unresolved external symbol "unsigned __int64 * _hb_CrapPool" (?_hb_CrapPool@@3PA_KA) referenced in function "struct hb_pair_t<float,struct Triple> & __cdecl Crap<struct hb_pair_t<float,struct Triple> >(void)" (??$Crap@U?$hb_pair_t@MUTriple@@@@@@YAAEAU?$hb_pair_t@MUTriple@@@@XZ)
hb-subset-instancer-solver.cc.obj : error LNK2001: unresolved external symbol "unsigned __int64 * _hb_CrapPool" (?_hb_CrapPool@@3PA_KA)
test-subset-instancer-solver.cc.obj : error LNK2019: unresolved external symbol "unsigned __int64 const * const _hb_NullPool" (?_hb_NullPool@@3QB_KB) referenced in function "public: static struct hb_pair_t<float,struct Triple> const & __cdecl Null<struct hb_pair_t<float,struct Triple> >::get_null(void)" (?get_null@?$Null@U?$hb_pair_t@MUTriple@@@@@@SAAEBU?$hb_pair_t@MUTriple@@@@XZ)
hb-subset-instancer-solver.cc.obj : error LNK2001: unresolved external symbol "unsigned __int64 const * const _hb_NullPool" (?_hb_NullPool@@3QB_KB)
src\test-instancer-solver.exe : fatal error LNK1120: 2 unresolved externals
clang: error: linker command failed with exit code 1120 (use -v to see invocation)
[8/13] Linking target src/test-multimap.exe
FAILED: src/test-multimap.exe src/test-multimap.pdb
"clang"  -Wl,/MACHINE:X64 -Wl,/OUT:src/test-multimap.exe src/test-multimap.exe.p/test-multimap.cc.obj src/test-multimap.exe.p/hb-static.cc.obj "-Wl,/nologo" "-Wl,/release" "-Wl,/nologo" "-Wl,/DEBUG" "-Wl,/PDB:src/test-multimap.pdb" "src/harfbuzz.lib" "subprojects/freetype-2.13.0/freetype.lib" "-Wl,/SUBSYSTEM:CONSOLE" "-lkernel32" "-luser32" "-lgdi32" "-lwinspool" "-lshell32" "-lole32" "-loleaut32" "-luuid" "-lcomdlg32" "-ladvapi32"
test-multimap.cc.obj : error LNK2019: unresolved external symbol "unsigned __int64 * _hb_CrapPool" (?_hb_CrapPool@@3PA_KA) referenced in function "unsigned int & __cdecl Crap<unsigned int>(void)" (??$Crap@I@@YAAEAIXZ)
test-multimap.cc.obj : error LNK2019: unresolved external symbol "unsigned __int64 const * const _hb_NullPool" (?_hb_NullPool@@3QB_KB) referenced in function "public: static unsigned int const & __cdecl Null<unsigned int>::get_null(void)" (?get_null@?$Null@I@@SAAEBIXZ)
src\test-multimap.exe : fatal error LNK1120: 2 unresolved externals
clang: error: linker command failed with exit code 1120 (use -v to see invocation)
[9/13] Linking target src/test-repacker.exe
FAILED: src/test-repacker.exe src/test-repacker.pdb
"clang"  -Wl,/MACHINE:X64 -Wl,/OUT:src/test-repacker.exe src/test-repacker.exe.p/test-repacker.cc.obj src/test-repacker.exe.p/hb-static.cc.obj src/test-repacker.exe.p/graph_gsubgpos-context.cc.obj "-Wl,/nologo" "-Wl,/release" "-Wl,/nologo" "-Wl,/DEBUG" "-Wl,/PDB:src/test-repacker.pdb" "src/harfbuzz.lib" "subprojects/freetype-2.13.0/freetype.lib" "-Wl,/SUBSYSTEM:CONSOLE" "-lkernel32" "-luser32" "-lgdi32" "-lwinspool" "-lshell32" "-lole32" "-loleaut32" "-luuid" "-lcomdlg32" "-ladvapi32"
test-repacker.cc.obj : error LNK2019: unresolved external symbol "unsigned __int64 * _hb_CrapPool" (?_hb_CrapPool@@3PA_KA) referenced in function "struct hb_user_data_array_t::hb_user_data_item_t & __cdecl Crap<struct hb_user_data_array_t::hb_user_data_item_t>(void)" (??$Crap@Uhb_user_data_item_t@hb_user_data_array_t@@@@YAAEAUhb_user_data_item_t@hb_user_data_array_t@@XZ)
graph_gsubgpos-context.cc.obj : error LNK2001: unresolved external symbol "unsigned __int64 * _hb_CrapPool" (?_hb_CrapPool@@3PA_KA)
test-repacker.cc.obj : error LNK2019: unresolved external symbol "unsigned __int64 const * const _hb_NullPool" (?_hb_NullPool@@3QB_KB) referenced in function "public: static struct hb_user_data_array_t::hb_user_data_item_t const & __cdecl Null<struct hb_user_data_array_t::hb_user_data_item_t>::get_null(void)" (?get_null@?$Null@Uhb_user_data_item_t@hb_user_data_array_t@@@@SAAEBUhb_user_data_item_t@hb_user_data_array_t@@XZ)
graph_gsubgpos-context.cc.obj : error LNK2001: unresolved external symbol "unsigned __int64 const * const _hb_NullPool" (?_hb_NullPool@@3QB_KB)
test-repacker.cc.obj : error LNK2019: unresolved external symbol "unsigned char const * const _hb_Null_OT_RangeRecord" (?_hb_Null_OT_RangeRecord@@3QBEB) referenced in function "public: static struct OT::Layout::Common::RangeRecord<struct OT::Layout::SmallTypes> const & __cdecl Null<struct OT::Layout::Common::RangeRecord<struct OT::Layout::SmallTypes> >::get_null(void)" (?get_null@?$Null@U?$RangeRecord@USmallTypes@Layout@OT@@@Common@Layout@OT@@@@SAAEBU?$RangeRecord@USmallTypes@Layout@OT@@@Common@Layout@OT@@XZ)
src\test-repacker.exe : fatal error LNK1120: 3 unresolved externals
clang: error: linker command failed with exit code 1120 (use -v to see invocation)
[10/13] Linking target src/test-map.exe
FAILED: src/test-map.exe src/test-map.pdb
"clang"  -Wl,/MACHINE:X64 -Wl,/OUT:src/test-map.exe src/test-map.exe.p/test-map.cc.obj src/test-map.exe.p/hb-static.cc.obj "-Wl,/nologo" "-Wl,/release" "-Wl,/nologo" "-Wl,/DEBUG" "-Wl,/PDB:src/test-map.pdb" "src/harfbuzz.lib" "subprojects/freetype-2.13.0/freetype.lib" "-Wl,/SUBSYSTEM:CONSOLE" "-lkernel32" "-luser32" "-lgdi32" "-lwinspool" "-lshell32" "-lole32" "-loleaut32" "-luuid" "-lcomdlg32" "-ladvapi32"
test-map.cc.obj : error LNK2019: unresolved external symbol "unsigned __int64 * _hb_CrapPool" (?_hb_CrapPool@@3PA_KA) referenced in function "struct hb_hashmap_t<unsigned int,unsigned int,1>::item_t & __cdecl Crap<struct hb_hashmap_t<unsigned int,unsigned int,1>::item_t>(void)" (??$Crap@Uitem_t@?$hb_hashmap_t@II$00@@@@YAAEAUitem_t@?$hb_hashmap_t@II$00@@XZ)
test-map.cc.obj : error LNK2019: unresolved external symbol "unsigned __int64 const * const _hb_NullPool" (?_hb_NullPool@@3QB_KB) referenced in function "public: static struct hb_hashmap_t<unsigned int,unsigned int,1>::item_t const & __cdecl Null<struct hb_hashmap_t<unsigned int,unsigned int,1>::item_t>::get_null(void)" (?get_null@?$Null@Uitem_t@?$hb_hashmap_t@II$00@@@@SAAEBUitem_t@?$hb_hashmap_t@II$00@@XZ)
test-map.cc.obj : error LNK2019: unresolved external symbol "unsigned int const minus_1" (?minus_1@@3IB) referenced in function "public: static unsigned int const & __cdecl hb_hashmap_t<unsigned int,unsigned int,1>::item_t::default_value<1,0>(void)" (??$default_value@$00$0A@@item_t@?$hb_hashmap_t@II$00@@SAAEBIXZ)
src\test-map.exe : fatal error LNK1120: 3 unresolved externals
clang: error: linker command failed with exit code 1120 (use -v to see invocation)
[11/13] Linking target src/test-bimap.exe
FAILED: src/test-bimap.exe src/test-bimap.pdb
"clang"  -Wl,/MACHINE:X64 -Wl,/OUT:src/test-bimap.exe src/test-bimap.exe.p/test-bimap.cc.obj src/test-bimap.exe.p/hb-static.cc.obj "-Wl,/nologo" "-Wl,/release" "-Wl,/nologo" "-Wl,/DEBUG" "-Wl,/PDB:src/test-bimap.pdb" "src/harfbuzz.lib" "subprojects/freetype-2.13.0/freetype.lib" "-Wl,/SUBSYSTEM:CONSOLE" "-lkernel32" "-luser32" "-lgdi32" "-lwinspool" "-lshell32" "-lole32" "-loleaut32" "-luuid" "-lcomdlg32" 
"-ladvapi32"
test-bimap.cc.obj : error LNK2019: unresolved external symbol "unsigned int const minus_1" (?minus_1@@3IB) referenced in function "public: static unsigned int const & __cdecl hb_hashmap_t<unsigned int,unsigned int,1>::item_t::default_value<1,0>(void)" (??$default_value@$00$0A@@item_t@?$hb_hashmap_t@II$00@@SAAEBIXZ)
test-bimap.cc.obj : error LNK2019: unresolved external symbol "unsigned __int64 * _hb_CrapPool" (?_hb_CrapPool@@3PA_KA) referenced in function "unsigned int & __cdecl Crap<unsigned int>(void)" (??$Crap@I@@YAAEAIXZ)
test-bimap.cc.obj : error LNK2019: unresolved external symbol "unsigned __int64 const * const _hb_NullPool" (?_hb_NullPool@@3QB_KB) referenced in function "public: static unsigned int const & __cdecl Null<unsigned int>::get_null(void)" (?get_null@?$Null@I@@SAAEBIXZ)
src\test-bimap.exe : fatal error LNK1120: 3 unresolved externals
clang: error: linker command failed with exit code 1120 (use -v to see invocation)
[12/13] Linking target src/test-vector.exe
FAILED: src/test-vector.exe src/test-vector.pdb
"clang"  -Wl,/MACHINE:X64 -Wl,/OUT:src/test-vector.exe src/test-vector.exe.p/test-vector.cc.obj src/test-vector.exe.p/hb-static.cc.obj "-Wl,/nologo" "-Wl,/release" "-Wl,/nologo" "-Wl,/DEBUG" "-Wl,/PDB:src/test-vector.pdb" "src/harfbuzz.lib" "subprojects/freetype-2.13.0/freetype.lib" "-Wl,/SUBSYSTEM:CONSOLE" "-lkernel32" "-luser32" "-lgdi32" "-lwinspool" "-lshell32" "-lole32" "-loleaut32" "-luuid" "-lcomdlg32" "-ladvapi32"
test-vector.cc.obj : error LNK2019: unresolved external symbol "unsigned __int64 * _hb_CrapPool" (?_hb_CrapPool@@3PA_KA) referenced in function "struct hb_user_data_array_t::hb_user_data_item_t & __cdecl 
Crap<struct hb_user_data_array_t::hb_user_data_item_t>(void)" (??$Crap@Uhb_user_data_item_t@hb_user_data_array_t@@@@YAAEAUhb_user_data_item_t@hb_user_data_array_t@@XZ)
test-vector.cc.obj : error LNK2019: unresolved external symbol "unsigned __int64 const * const _hb_NullPool" (?_hb_NullPool@@3QB_KB) referenced in function "public: static struct hb_user_data_array_t::hb_user_data_item_t const & __cdecl Null<struct hb_user_data_array_t::hb_user_data_item_t>::get_null(void)" (?get_null@?$Null@Uhb_user_data_item_t@hb_user_data_array_t@@@@SAAEBUhb_user_data_item_t@hb_user_data_array_t@@XZ)
src\test-vector.exe : fatal error LNK1120: 2 unresolved externals
clang: error: linker command failed with exit code 1120 (use -v to see invocation)
[13/13] Linking target src/test-tuple-varstore.exe
FAILED: src/test-tuple-varstore.exe src/test-tuple-varstore.pdb
"clang"  -Wl,/MACHINE:X64 -Wl,/OUT:src/test-tuple-varstore.exe src/test-tuple-varstore.exe.p/test-tuple-varstore.cc.obj src/test-tuple-varstore.exe.p/hb-subset-instancer-solver.cc.obj src/test-tuple-varstore.exe.p/hb-static.cc.obj "-Wl,/nologo" "-Wl,/release" "-Wl,/nologo" "-Wl,/DEBUG" "-Wl,/PDB:src/test-tuple-varstore.pdb" "src/harfbuzz.lib" "subprojects/freetype-2.13.0/freetype.lib" "-Wl,/SUBSYSTEM:CONSOLE" "-lkernel32" "-luser32" "-lgdi32" "-lwinspool" "-lshell32" "-lole32" "-loleaut32" "-luuid" "-lcomdlg32" "-ladvapi32"
test-tuple-varstore.cc.obj : error LNK2019: unresolved external symbol "unsigned __int64 const * const _hb_NullPool" (?_hb_NullPool@@3QB_KB) referenced in function "public: static struct OT::TupleVariationData const & __cdecl Null<struct OT::TupleVariationData>::get_null(void)" (?get_null@?$Null@UTupleVariationData@OT@@@@SAAEBUTupleVariationData@OT@@XZ)
hb-subset-instancer-solver.cc.obj : error LNK2001: unresolved external symbol "unsigned __int64 const * const _hb_NullPool" (?_hb_NullPool@@3QB_KB)
test-tuple-varstore.cc.obj : error LNK2019: unresolved external symbol "unsigned __int64 * _hb_CrapPool" (?_hb_CrapPool@@3PA_KA) referenced in function "struct hb_hashmap_t<struct hb_vector_t<bool,0> const *,struct hb_array_t<char const >,0>::item_t & __cdecl Crap<struct hb_hashmap_t<struct hb_vector_t<bool,0> const *,struct hb_array_t<char const >,0>::item_t>(void)" (??$Crap@Uitem_t@?$hb_hashmap_t@PEBU?$hb_vector_t@_N$0A@@@U?$hb_array_t@$$CBD@@$0A@@@@@YAAEAUitem_t@?$hb_hashmap_t@PEBU?$hb_vector_t@_N$0A@@@U?$hb_array_t@$$CBD@@$0A@@@XZ)
hb-subset-instancer-solver.cc.obj : error LNK2001: unresolved external symbol "unsigned __int64 * _hb_CrapPool" (?_hb_CrapPool@@3PA_KA)
test-tuple-varstore.cc.obj : error LNK2019: unresolved external symbol "unsigned int const minus_1" (?minus_1@@3IB) referenced in function "public: static unsigned int const & __cdecl hb_hashmap_t<unsigned int,unsigned int,1>::item_t::default_value<1,0>(void)" (??$default_value@$00$0A@@item_t@?$hb_hashmap_t@II$00@@SAAEBIXZ)
src\test-tuple-varstore.exe : fatal error LNK1120: 3 unresolved externals
clang: error: linker command failed with exit code 1120 (use -v to see invocation)
ninja: build stopped: subcommand failed.
```

</details>

Reply:
Weird. That symbol is declared in `hb-static.cc`...